### ISSUE TITLE
helm dep update fails to download chart

### ISSUE THREAD

**Issue by prydonius** [kubernetes/helm#1845]
In the latest canary build, `helm dep update` is broken, failing to download the chart.

cc @viglesiasce

```
$ helm dep update
Hang tight while we grab the latest from your chart repositories...
...Unable to get an update from the "local" chart repository (http://127.0.0.1:8879/charts):
    Get http://127.0.0.1:8879/charts/index.yaml: dial tcp 127.0.0.1:8879: getsockopt: connection refused
...Successfully got an update from the "stable" chart repository
...Successfully got an update from the "bitnami" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 2 charts
Downloading redis from repo https://kubernetes-charts.storage.googleapis.com/
Error: could not download https://kubernetes-charts.storage.googleapis.com/redis-0.4.2.tgz: repo with URL https://kubernetes-charts.storage.googleapis.com not found
```

**Comment by prydonius** [comment]
This seems to be fixed by removing the helm dir `rm ~/.helm` and re-initing, but then I ran into an even weirder issue in #1846.

What changed that requires re-initing?

**Comment by technosophos** [comment]
I'll be looking into these two today.

**Comment by technosophos** [comment]
Did you happen to manually change the URL for the chart repo?

**Comment by prydonius** [comment]
Initially we were getting this error in the CI with the old URL (http://storage.googleapis.com/kubernetes-charts), so we thought there was an issue with that URL and switched over to the subdomain version (https://kubernetes-charts.storage.googleapis.com/). We did this by manually changing the URLs in the existing index, and updating the CI scripts to use the new URL (https://github.com/kubernetes/charts/commit/c26821c495b4deb491c651302724f223e9e71a5f).

Although we stopped seeing the issue in the CI (I suspect this is because the jobs were running a clean workspace), I was able to locally reproduce it with the new URL too.

Both URLs are valid, so I'm not sure why it's failing to download them. It also looks like a `helm repo update` is run during `helm dep update` which may rule out any local changes to the index?
